src/components/form/components/input/checkbox.tsx

```diff
@@ -16,6 +16,8 @@ export function Checkbox(props: CheckboxProps) {
   function onChange(event: React.ChangeEvent<HTMLInputElement>) {
     const checked = event.target.checked;
     field.handleChange(checked);
+
+    return field.handleBlur();
```

Comment:
Pretending that the field blurred doesn't seem like a great way to handle the issue.

Either checkbox fields should be configured to revalidate on change, or we can change the global revalidation strategy to be on change for all types of inputs after the initial submit. I'd be in favor of the latter (change the global strategy).

Reply:
[I made a major edit to my comment above]